tpm/aziot-tpm-rs/src/lib.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+//! Idiomatic Rust bindings to the `aziot-tpm` C library.
+
+#![deny(rust_2018_idioms, warnings)]
+#![deny(clippy::all, clippy::pedantic)]
+#![allow(clippy::missing_errors_doc, clippy::module_name_repetitions)]
+#![deny(missing_docs)]
+
+mod error;
+mod tpm;
+
+pub use crate::error::Error;
+pub use crate::tpm::{Tpm, TpmDigest, TpmKey, TpmKeys};
+
+// Traits
```

Comment:
You don't need to replicate this from libiothsm. Two traits is unnecessary when one will suffice, and one trait is unnecessary when there's only one impl. Just make them crate-level fns.

Reply:
Yeah, I was debating about whether or not to leave the two traits as separate entities. One should do.

That said, I think it makes sense to use a Trait, as it'll make testing/mocking easier.

Reply:
As we'd discussed, if you mock the crate API, then there's no point writing such a test, because the API that has been mocked was the only thing worth testing.

We can test with `ibmswtpm2` and run the real code against it.

Reply:
My one concern with that approach is that if we parallelize the tests, we might run into problems where one test will register a key, and before it has a chance to sign something, another test registers another key, leading to weird shenanigans.

Then again, I've never actually written a proper end-to-end test in our repo, so I'm not sure how in-depth we'd go.

I'm cool with nixing the trait, just lmk

Reply:
Any tests involving `ibmswtpm2` will have to be serialized, as we'd already discussed.

For E2E tests they wouldn't be run in parallel anyway, because they all operate on global resources. The first step of an E2E test is installing the deb / rpm package and setting up the config for the test.

Reply:
Alrighty then, I shall make the traits b e g o n e